infra/experimental/sanitizers/ExecSan/execSan.cpp

```diff
@@ -60,6 +62,86 @@
 const std::string kTripWire = "/tmp/tripwire";
 // The PID of the root process we're fuzzing.
 pid_t g_root_pid;
+// Assuming the longest pathname is "/bin/bash".
+const int kShellPathnameLength = 10;
+// Assuming the syntax error pattern is
+//   within the first 100 chars of the write buffer.
+const int kErrorMessageLength = 100;
+// Shell injection bug confirmed with /tmp/tripwire.
+const int kShellInjection = 1;
+// Shell corruption bug speculated based on syntax error.
+const int kShellCorruption = 2;
+// Shells used by Processes.
+std::map<pid_t, std::string> pidShellMap;
+
+// Two kinds of bugs to detect.
+std::map<int, std::string> kBugMessageMap = {
+  {kShellInjection,  "Shell injection"},
+  {kShellCorruption, "Shell corruption"},
+};
+
+// Shells to inspect.
+std::set <std::string> kShellSet = {
+  "sh",
+  "bash",
+  "csh",
+  "dash",
+  "zsh"
+};
+
+// Syntax error messages of each shell.
+std::map<std::string, std::set<std::string>> kShellSytaxErrorMap = {
```

Comment:
typo in the name. Can you rename this `kShellSyntaxErrors` ? 

Also can you make this `const` ?